### PR TITLE
Feature/erc 02 Home hero

### PR DESCRIPTION
**Pull request by Melisa-izaquita**
# FooCamp - Escuela de Robotica

## Trello ticket
[Ticket Link](https://trello.com/c/FcUXOy7G/14-user-story-02-homehero)

## Description
The slider component was made that contains an image, title and text
393F0FDC-3DDB-4714-BEA0-10255FCB90F6

## To reproduce
Provide clear instructions to check the new feature (or to reproduce the issue).

1. Run the project with npm run dev
2. Open a page (http://localhost:8080/melisa-page.html)
3. See the component works as expected

## Screenshots
Put here the screenshots of the developed functionality (or fixed issues). For each viewport (if applies).

### CODE REVIEW

File: src/scss/components/_heroSlide.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
+.slider{
+  margin: 0 auto;
+  position: relative;
+  max-width: 1440px;
```
Comment (diegoamt) [inline src/scss/components/_heroSlide.scss:4]: Remove these properties: `margin`, `position` and `max-width` and add the class `component` in the template.

File: src/scss/components/_heroSlide.scss [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  &__img{
+    display: flex;
+    flex-direction: row-reverse;
+  }
+  
+  &__image{
+    border-radius: 250px 0px 0px 250px;
```
Comment (diegoamt) [inline src/scss/components/_heroSlide.scss:11]: use just `0` instead of `0px`

File: src/scss/components/_heroSlide.scss [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  }
+  &__content{
+    position: absolute;
+    max-width: 48%;
+    top: 118px;
+    margin-left: 80px;
+  }
+  &__title{
+    font-size: 48px;
+    color: #577598;
```
Comment (diegoamt) [inline src/scss/components/_heroSlide.scss:24]: create a variable for this color in `variables.scss`

File: src/scss/components/_heroSlide.scss [hunk 4]
```diff
@@ -0,0 +1,40 @@
+.slider{
+  margin: 0 auto;
+  position: relative;
+  max-width: 1440px;
+  &__img{
+    display: flex;
+    flex-direction: row-reverse;
```
Comment (diegoamt) [inline src/scss/components/_heroSlide.scss:7]: add `width: 100%;` to make sure this element will expand across all the container.

File: src/scss/main.scss [hunk 5]
```diff
@@ -13,3 +13,4 @@
 
 // Components
 @import 'components/component';
+@import 'components/heroSlide';
```
Comment (diegoamt) [inline src/scss/main.scss:16]: rename this file to `hero-slide.scss`

File: templates/heroSlide.hbs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+<div  class="slider">
+  <div class="slider__content">
+    <h2 class="slider__title">{{title}}</h2>
+    <p class="slider__paragraph">{{description}}</p>
+  </div>
+  <div class="slider__img">
+    <img class="slider__image" src="{{{assetSrc image}}}" alt="{{alttext}}">
```
Comment (diegoamt) [inline templates/heroSlide.hbs:7]: The ID of the alt text field is wrong. Please use `altText` instead of alttext.

**Review by diegoamt — CHANGES_REQUESTED** [review]

File: create-env.js [hunk 7]
```diff
@@ -4,3 +4,6 @@ fs.copyFile('.env.example', '.env', (err) => {
   if (err) throw err;
   console.log('.env was created successfully');
 });
+const Handlebars = require("handlebars");
```
Comment (camilomejiag) [inline create-env.js:7]: Por que agregaste Handlebars aca? Veo que solo lo utilizas para el console log de abajo. El proyecto tiene un setup ya para poder mostrar todos los .hbs
Si quitas esto (y la dependencia del package.json) deberia funcionarte, no?

File: templates/page.hbs [hunk 8]
```diff
@@ -23,6 +23,14 @@
 
     {{{ module footer }}}
 
+    <main>
+      {{#each components}}
+        {{{module this}}}
+      {{/each }}
+    </main>
```
Comment (diegoamt) [inline templates/page.hbs:30]: remove this extra main element. (maybe was added in the automatic merge resolving git conflicts)

File: create-env.js (hunk 7 above)
Comment (diegoamt) [reply create-env.js:7]: agree, remove it

**Review by diegoamt — CHANGES_REQUESTED** [review]

File: templates/heroSlide.hbs [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        <img class="slider__image" src="{{{assetSrc image}}}" alt="{{altText}}">
+      </div>
+    </div>
+  </div>
+  <div class="dots-list">
+    <button class="dot-button active"></button>
+    <button class="dot-button"></button>
+  </div>
+</div>
+
```
Comment (diegoamt) [inline templates/heroSlide.hbs:27]: The template `heroSlide.hbs` should be only used for one slide. The logic for the carousel should be in a new remplate called `hero.hbs`. Use the `{{#each ..}}` expression along with the helper `{{{module ...}}}` to render the slides.

**Review by diegoamt — CHANGES_REQUESTED** [review]

**Review by camilomejiag — APPROVED** [review]

**Pull request merged**